### PR TITLE
Enable getWalletCollateral with KeyWallet

### LINKED ISSUE #856: Use collateral selection in `getWalletCollateral`
we use collateral selection in `utxosAt`, but not in `getWalletCollateral`

https://github.com/Plutonomicon/cardano-transaction-lib/blob/be0c81b6dde21fa5370740b642aa549e93f74048/src/QueryM.purs#L441

### PR DESCRIPTION
**Pull request by klntsky**
Closes #856 

### Pre-review checklist

- [x] All code has been formatted using our config (`make format`)
- [x] The integration and unit tests have been run (`npm run test`) locally
- [x] The changelog has been updated under the `## Unreleased` header, using the appropriate sub-headings (`### Added`, `### Removed`, `### Fixed`), and the links to the appropriate issues/PRs have been included

### CODE REVIEW

File: test/Plutip.purs [hunk 1]
```diff
@@ -156,7 +161,17 @@ suite = do
...
       runPlutipContract config distribution \(alice /\ bob) -> do
         withKeyWallet alice do
-          pure unit -- sign, balance, submit, etc.
+          getWalletCollateral >>= liftEffect <<< case _ of
+            Nothing -> throw "Unable to get collateral"
+            Just
+              [ TransactionUnspentOutput
+                  { output: TransactionOutput { amount } }
+              ] -> do
+              when (amount /= lovelaceValueOf (BigInt.fromInt 1_000_000_000)) do
```
Comment (ngua) [inline test/Plutip.purs:170]:
```suggestion
              unless (amount == lovelaceValueOf (BigInt.fromInt 1_000_000_000)) do
```

**Review by ngua — COMMENTED** [review]
Just one small thing, otherwise LGTM

File: src/BalanceTx/BalanceTx.purs [hunk 2]
```diff
@@ -168,8 +167,7 @@ instance Show GetWalletAddressError where
   show = genericShow
 
 data GetWalletCollateralError
-  = CannotRequestCollateralForWallet ImpossibleError
-  | CouldNotGetCollateral
+  = CouldNotGetCollateral
   | WalletNotSpecified
```
Comment (errfrom) [inline src/BalanceTx/BalanceTx.purs:171]: `WalletNotSpecified` can be removed

**Review by errfrom — APPROVED** [review]
LGTM

**Review by ngua — APPROVED** [review]

**Pull request merged**